pkg/build/builder/cmd/builder.go

```diff
@@ -41,22 +41,28 @@ func run(builderFactory factoryFunc) {
 		authcfg     docker.AuthConfiguration
 		authPresent bool
 	)
+	output := true
 	if len(build.Parameters.Output.DockerImageReference) == 0 {
 		if build.Parameters.Output.To != nil {
 			log.Fatalf("Cannot determine an output image reference. Make sure a registry service is running.")
 		}
-		log.Fatal("Build output has an empty Docker image reference.")
+		output = false
 	}
-	registry, _, _, _, err := image.SplitDockerPullSpec(build.Parameters.Output.DockerImageReference)
-	if err != nil {
-		log.Fatalf("Build output does not have a valid Docker image reference: %v", err)
+	if output {
+		registry, _, _, _, err := image.SplitDockerPullSpec(build.Parameters.Output.DockerImageReference)
+		if err != nil {
+			log.Fatalf("Build output does not have a valid Docker image reference: %v", err)
+		}
+		authcfg, authPresent = dockercfg.NewHelper().GetDockerAuth(registry)
 	}
-	authcfg, authPresent = dockercfg.NewHelper().GetDockerAuth(registry)
-
 	b := builderFactory(client, endpoint, authcfg, authPresent, &build)
 	if err = b.Build(); err != nil {
 		log.Fatalf("Build error: %v", err)
 	}
+	if !output {
+		log.Print("[WARNING] Build does not have an Output defined, no output image was pushed to a registry.")
```

Comment:
i don't love it, but it's the only one so far, doesn't seem worth switching.  i assume there's a reason this didn't use glog in the first place...